### PR TITLE
[Feature] Add `preferredCameraIndex` for picking

### PR DESCRIPTION
**Pull request by ihpcoder**
添加一个初始化参数initCameraIndex，可以创建时选择摄像头（后置摄像头）创建

### CODE REVIEW

File: lib/src/widget/camera_picker.dart [hunk 1]
```diff
@@ -356,7 +363,7 @@ class CameraPickerState extends State<CameraPicker>
 
   /// A getter to the current [CameraDescription].
   /// 获取当前相机实例
-  CameraDescription get currentCamera => cameras.elementAt(currentCameraIndex);
+  CameraDescription get currentCamera => cameras.elementAt(currentCameraIndex >= cameras.length ? cameras.length-1 : currentCameraIndex);
```
Comment (AlexV525) [inline lib/src/widget/camera_picker.dart:366]: 此处不存在超出的情况，为什么要判断？

File: lib/src/widget/camera_picker.dart [hunk 2]
```diff
@@ -501,6 +508,13 @@ class CameraPickerState extends State<CameraPicker>
       // time initializing cameras, so available cameras should be fetched.
       if (cameraDescription == null) {
         cameras = await availableCameras();
+        if(cameras.isNotEmpty) {
+          if (cameras.length <= widget.initCameraIndex) {
+            currentCameraIndex = cameras.length - 1;
```
Comment (AlexV525) [inline lib/src/widget/camera_picker.dart:513]: 这里有明显的争议。如果我随便传入了5，而实际上只有2，为什么需要是最后一个而不能是第一个？

File: lib/src/widget/camera_picker.dart [hunk 3]
```diff
@@ -555,7 +569,7 @@ class CameraPickerState extends State<CameraPicker>
   /// 按顺序切换相机。当达到相机数量时从头开始。
   void switchCameras() {
     ++currentCameraIndex;
-    if (currentCameraIndex == cameras.length) {
+    if (currentCameraIndex >= cameras.length) {
```
Comment (AlexV525) [inline lib/src/widget/camera_picker.dart:572]: 理论上不成立。

**Comment by ihpcoder** [comment]
大意了，我在修改下



---Original---
From: "Alex ***@***.***&gt;
Date: Fri, Aug 13, 2021 15:43 PM
To: ***@***.***&gt;;
Cc: ***@***.******@***.***&gt;;
Subject: Re: [fluttercandies/flutter_wechat_camera_picker] add initCameraIndex (#43)




 
@AlexV525 commented on this pull request.
 
 
In lib/src/widget/camera_picker.dart:
 &gt; @@ -356,7 +363,7 @@ class CameraPickerState extends State<CameraPicker&gt;      /// A getter to the current [CameraDescription].    /// 获取当前相机实例 -  CameraDescription get currentCamera =&gt; cameras.elementAt(currentCameraIndex); +  CameraDescription get currentCamera =&gt; cameras.elementAt(currentCameraIndex &gt;= cameras.length ? cameras.length-1 : currentCameraIndex);  
此处不存在超出的情况，为什么要判断？
 
 
In lib/src/widget/camera_picker.dart:
 &gt; @@ -501,6 +508,13 @@ class CameraPickerState extends State<CameraPicker&gt;        // time initializing cameras, so available cameras should be fetched.        if (cameraDescription == null) {          cameras = await availableCameras(); +        if(cameras.isNotEmpty) { +          if (cameras.length <= widget.initCameraIndex) { +            currentCameraIndex = cameras.length - 1;  
这里有明显的争议。如果我随便传入了5，而实际上只有2，为什么需要是最后一个而不能是第一个？
 
 
In lib/src/widget/camera_picker.dart:
 &gt; @@ -555,7 +569,7 @@ class CameraPickerState extends State<CameraPicker&gt;    /// 按顺序切换相机。当达到相机数量时从头开始。    void switchCameras() {      ++currentCameraIndex; -    if (currentCameraIndex == cameras.length) { +    if (currentCameraIndex &gt;= cameras.length) {  
理论上不成立。
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by AlexV525** [comment]
有自测过吗？

**Review by AlexV525 — COMMENTED** [review]
合并前我需要仔细考虑一下它的影响，预计在下周三前尽早给出一个明确的答复。

**Comment by ihpcoder** [comment]
好的



发自我的iPhone


------------------ Original ------------------
From: Alex Li ***@***.***&gt;
Date: Fri,Aug 13,2021 3:56 PM
To: fluttercandies/flutter_wechat_camera_picker ***@***.***&gt;
Cc: ihpcoder ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [fluttercandies/flutter_wechat_camera_picker] add initCameraIndex (#43)





 
@AlexV525 commented on this pull request.
 
合并前我需要仔细考虑一下它的影响，预计在下周三前尽早给出一个明确的答复。
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by AlexV525** [comment]
根据 CI 先调整格式问题。

**Comment by AlexV525** [comment]
根据 `CameraDescription.lensDirection` 来看，可以直接筛选出后置摄像头。建议将实现改为对应筛选。

**Comment by ihpcoder** [comment]
回头看下



发自我的iPhone


------------------ Original ------------------
From: Alex Li ***@***.***&gt;
Date: Wed,Sep 1,2021 10:22 AM
To: fluttercandies/flutter_wechat_camera_picker ***@***.***&gt;
Cc: ihpcoder ***@***.***&gt;, Author ***@***.***&gt;
Subject: Re: [fluttercandies/flutter_wechat_camera_picker] [Feature] Add `preferredCameraIndex` for picking (#43)





 
根据 CameraDescription.lensDirection 来看，可以直接筛选出后置摄像头。建议将实现改为对应筛选。
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by AlexV525** [comment]
Closing in flavor of #57 .

**Pull request closed without merging**